blockchain/blocks/src/header.rs

```diff
@@ -301,6 +305,31 @@ impl BlockHeader {
 
         Ok(())
     }
+    /// Returns true if (h(vrfout) * totalPower) < (e * sectorSize * 2^256)
+    pub fn is_ticket_winner(&self, mpow: BigUint, net_pow: BigUint) -> bool {
+        /*
+        Need to check that
+        (h(vrfout) + 1) / (max(h) + 1) <= e * myPower / totalPower
+        max(h) == 2^256-1
+        which in terms of integer math means:
+        (h(vrfout) + 1) * totalPower <= e * myPower * 2^256
+        in 2^256 space, it is equivalent to:
+        h(vrfout) * totalPower < e * myPower * 2^256
+        */
+
+        // TODO switch ticket for election_proof
+        let h = sha2::Sha256::digest(self.ticket.vrfproof.bytes());
+        let mut lhs = BigUint::from_bytes_le(&h);
+        lhs *= net_pow;
+
+        // rhs = sectorSize * 2^256
+        // rhs = sectorSize << 256
+        let mut rhs = mpow << SHA_256_BITS;
+        rhs *= BigUint::from(BLOCKS_PER_EPOCH);
+
+        // h(vrfout) * totalPower < e * sectorSize * 2^256
+        lhs.cmp(&rhs) < Ordering::Equal
```

Comment:
```suggestion
        lhs < rhs
```
lol

Reply:
🤔

Reply:
haha I always like doing things the hard way 🙃